versions/ufo3/lib.plist.md

```diff
@@ -41,6 +41,19 @@ This defines a preferred glyph name to Postscript glyph name mapping for glyphs
 
 The mapping is stored as a dictionary with glyphs names as keys and Postscript glyph names as values. Both keys and values must be strings. The values must conform to the Postscript glyph naming specification. The dictionary may contain glyph names that are not in the font. The dictionary may not contain a key, value pair for all glyphs in the font. If a glyph's name is not defined in this mapping, the glyph's name should be used as the Postscript name.
 
+#### public.skipExportGlyphs
+
+This key is a list of glyph names used for representing glyphs that the user does not want exported to the final font file. The UFO compiler is expected to:
+
+1. Remove these glyphs before the compilation run.
+2. Decompose the listed glyphs everywhere they are used as components.
+3. Prune all groups of the listed glyphs, possibly leaving empty groups.
```

Comment:
Empty groups really makes me nervous. Can we specify that any empty groups will be removed? I'm not sure if this would have var font related blowback, but leaving empty groups seems really wrong.

Reply:
If the numbers in this list indicate some sort ordered steps, then `step 2, Decompose..` needs to happen **before** any glyphs are removed from the ufo.

Reply:
+1 on flipping the order of 1 and 2. It doesn't really matter, but it's a more algorithmically sound order.

I'm ambivalent about empty groups. The only place that they could be a problem is in kerning, right? Empty groups are legal in the current spec, so I don't feel strongly that we need to say anything about it here. If we do want to say something, perhaps the best place would be in the `kerning.plist` specification. Something like "Empty groups may be referenced by kerning pairs. It is the responsibility of the authoring tool to remove these from compiled fonts when needed." or something to that effect.

Reply:
I agree with Tal. The UFO file itself should be able to store incomplete data. There is responsibility for the authoring tool (or compiler) to warn, repair or ignore.

Reply:
My reading of this is that the pruning of empty groups (step 4) is only for kerning, and isn't changing the UFO, just that the UFO complier needs to do that so that it doesn't build kerning with empty groups from `kerning.plist`

Reply:
So, should I remove the mentioning of empty groups and add Tal's sentence to the kerning specification? It's basically a reminder to implementers.

FWIW, ufo2ft has been skipping empty groups since before this change.

Reply:
@madig Sounds good.